newm-server/src/main/kotlin/io/newm/server/features/marketplace/database/MarketplaceSaleEntity.kt

```diff
@@ -58,34 +62,36 @@ class MarketplaceSaleEntity(id: EntityID<UUID>) : UUIDEntity(id) {
             costPolicyId = costPolicyId,
             costAssetName = costAssetName,
             costAmount = costAmount,
+            costAmountUsd = costAmountUsd,
             maxBundleSize = maxBundleSize,
             totalBundleQuantity = totalBundleQuantity,
             bundleAmount = bundleAmount,
             availableBundleQuantity = availableBundleQuantity,
             song =
-                SongEntity[songId].run {
-                    val release = ReleaseEntity[releaseId!!]
-                    Sale.Song(
-                        id = id.value,
-                        artistId = ownerId.value,
-                        artistName = UserEntity.findById(ownerId)?.stageOrFullName,
-                        title = title,
-                        genres = genres.toList(),
-                        moods = moods?.toList(),
-                        coverArtUrl = release.coverArtUrl,
-                        clipUrl = clipUrl,
-                        tokenAgreementUrl = tokenAgreementUrl,
-                        collaborators =
-                            CollaborationEntity.findBySongId(songId.value).map {
-                                Sale.SongCollaborator(
-                                    id = it.id.value,
-                                    name = UserEntity.getByEmail(it.email)?.stageOrFullName,
-                                    role = it.role
-                                )
-                            }
-                    )
-                }
+                Sale.Song(
+                    id = songId.value,
+                    artistId = artist.id.value,
+                    artistName = artist.stageOrFullName,
+                    artistPictureUrl = artist.pictureUrl,
+                    title = song.title,
+                    description = song.description,
+                    parentalAdvisory = song.parentalAdvisory,
+                    genres = song.genres.toList(),
+                    moods = song.moods?.toList(),
+                    coverArtUrl = release.arweaveCoverArtUrl?.arweaveToWebUrl(),
```

Comment:
Can we get a url from nftcdn.io for this instead of hammering arweave hosts directly?

Reply:
I'll create a separate task for that.